chain-core/src/init/coin.rs

```diff
@@ -18,7 +18,7 @@ use std::convert::TryFrom;
 use std::{fmt, mem, ops, result, slice};
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Encode)]
-pub struct Coin(u64);
+pub struct Coin(pub u64);
```

Comment:
preferable this wouldn't be exposed, as it'd allow changing the internal value that would break Coin's invariants (e.g. one could make it more than the maximum supply).

it can be kept as `pub struct Coin(u64);` -- if you need the internal u64, just use `.into()` / `u64::from(...)`

Reply:
ok